### DISCUSSION TITLE
Drawing bond order

### DISCUSSION THREAD

**Discussion by kystyn** [rdkit/rdkit#4824] [Q&A] [answered]
Hello!
First of all, I am not chemist, just software engineer, so if I use wrong terms do not hesitate to tell me about it :)
My question is devoted to aromatic bonds drawing. I use MolDraw2DSVG and I am interested if there is a way how to receive information about order of **drawn** bond between concrete CH*-atoms in aromatic atom. For example, if we consider benzene, there are 3 bonds of second order and 3 bonds of first order and they must alternate. And there is two variants how to draw it. It is important for me to receive information about the way you draw it. The only way I found is to generate metadata for SVG and get this info from _bond-smiles_ attribute of bond. Is there more elegant method?
Thanks!

**Comment by greglandrum** [comment, marked as answer]
Hi @kystyn,
By default aromatic rings are represented in the RDKit with aromatic bonds between the atoms. You can see that here:
```
In [9]: m = Chem.MolFromSmiles('Cc1ncc(C)cc1')

In [10]: m.GetBondBetweenAtoms(1,2).GetBondType()
Out[10]: rdkit.Chem.rdchem.BondType.AROMATIC

In [11]: m.GetBondBetweenAtoms(2,3).GetBondType()
Out[11]: rdkit.Chem.rdchem.BondType.AROMATIC
```
Because chemists generally don't like to see aromatic bonds in drawings, the drawing code "Kekulizes" the molecule: converting aromatic bonds to sequences of alternating single and double bonds. As you point out, there are multiple ways to Kekulize any given molecule. When drawing the molecule the Kekulization is done on a copy of the molecule, so you can't directly determine which form was selected. However, you can do the Kekulization manually and then just interrogate the molecule about the bond types:
```
In [12]: Chem.Kekulize(m)

In [13]: m.GetBondBetweenAtoms(1,2).GetBondType()
Out[13]: rdkit.Chem.rdchem.BondType.DOUBLE

In [14]: m.GetBondBetweenAtoms(2,3).GetBondType()
Out[14]: rdkit.Chem.rdchem.BondType.SINGLE
```
If you pass this pre-Kekulized molecule to the drawing code, it won't repeat the Kekulization (actually it will, but the second Kekulization step won't change anything) and will draw the bonds as set in the molecule.


A question for you: why do you need this information? There may be an easier way to get it.

**Reply by kystyn** [reply]
Hi @greglandrum greglandrum. Thank you for your answer. That's exactly what I wanted to see.
I need this information because I participate in project devoted to chemical images recognition and translation of them into InChi format. Project is inspired with Kaggle competition https://www.kaggle.com/c/bms-molecular-translation/

**Comment by kystyn** [comment]
As we started this conversation I would like to ask you one more question. Is there any way to get information about drawing of atom labels? I found that I can get atom labels (C, N, O, ...) and Hs. So I can understand, for example, that if atom label is _O_ and _explicitHs == 1_ then _OH_ radical should be drown. But there are two ways to draw this radical: either OH, or HO. And I would like to know which is used, if it is possible

**Comment by DavidACosgrove** [comment]
Hi,
Whether OH or HO is drawn will depend on the rest of the molecule. The bond
will always go to the O.  This is most easily demonstrated with a picture
but I don’t have my computer with me at the moment. I’ll post one in due
course.
Dave


On Mon, 27 Dec 2021 at 11:12, Konstantin Damaskinskiy <
***@***.***> wrote:

> As we started this conversation I would like to ask you one more question.
> Is there any way to get information about drawing of atom labels? I found
> that I can get atom labels (C, N, O, ...) and Hs. So I can understand, for
> example, that if atom label is *O* and *explicitHs == 1* then *OH*
> radical should be drown. But there are two ways to draw this radical:
> either OH, or HO. And I would like to know which is used, if it is possible
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/rdkit/rdkit/discussions/4824#discussioncomment-1874643>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACGF2FUKRNOUFFOO6LNCBUDUTBC2LANCNFSM5KUSVE3A>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>
-- 
David Cosgrove
Freelance computational chemistry and chemoinformatics developer
http://cozchemix.co.uk

**Comment by DavidACosgrove** [comment]
@kystyn This shows how the OH will be drawn in most cases:
![image](https://user-images.githubusercontent.com/9198870/147556703-6894ec37-832b-4f7d-aaf4-2fd8bbc6b896.png)
and this shows that N is a bit more complicated:
![image](https://user-images.githubusercontent.com/9198870/147556993-0b118e0a-f8b6-4be4-95e4-82d73e05a72a.png)

Is that the sort of thing you were looking for?

**Reply by kystyn** [reply]
@DavidACosgrove Thank you for your answer! Nevertheless, I was looking for some function that would return positions of atomic labels or, at least, the drawn text: perfectly, I want to receive a structure like: {atom: <rdkit.Chem.rdchem.Atom>, text: 'H^+\\N^2', textPosition: [x, y]}.
